ql/termstructures/volatility/optionlet/optionletstripper.cpp

```diff
@@ -47,7 +50,7 @@ namespace QuantLib {
         registerWith(discount_);
         registerWith(Settings::instance().evaluationDate());
 
-        Period indexTenor = iborIndex_->tenor();
+        Period indexTenor = optionletFrequency_ ? *optionletFrequency_ : iborIndex_->tenor();
```

Comment:
Thanks, Paolo.  For further safety I'd add something like
```
if (ext::dynamic_pointer_cast<OvernightIndex>(iborIndex_) {
    QL_REQUIRE(optionletFrequency_, "an optionlet frequency is required when using an overnight index");
}
```
otherwise it would still be possible to pass an overnight and not specify the frequency.